src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -0,0 +1,228 @@
+                    _writtenViewComponents.Add(descriptorName);
+                    WriteClass(descriptor);
+                }
+            }
+        }
+
+        private void WriteClass(TagHelperDescriptor descriptor)
+        {
+            // Add target element.
+            BuildTargetElementString(descriptor);
+
+            // Initialize declaration.
+            var tagHelperTypeName = $"{_context.TagHelpersNamespace}.{nameof(TagHelper)}";
+
+            var shortName = descriptor.PropertyBag[ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey];
+            var className = $"__Generated__{shortName}ViewComponentTagHelper";
+
+            using (Writer.BuildClassDeclaration("public", className, new[] { tagHelperTypeName }))
+            {
+                // Add view component helper for reasons.
+                Writer.WriteVariableDeclaration(
+                    $"private readonly {_context.IViewComponentHelperType}",
+                    $"_viewComponentHelper", "");
+
+                // Add constructor.
+                BuildConstructorString(className);
+
+                // Add attributes.
+                BuildAttributeDeclarations(descriptor);
+
+                // Add process method.
+                BuildProcessMethodString(descriptor);
+            }
+        }
+
+        private void BuildConstructorString(string className)
+        {
+            KeyValuePair<string, string> helperPair = new KeyValuePair<string, string>(
+                _context.IViewComponentHelperType,
+                "viewComponentHelper");
+
+            using (Writer.BuildConstructor(
+                "public",
+                className,
+                new List<KeyValuePair<string, string>>() { helperPair }))
+            {
+                Writer.WriteVariableDeclaration("", "_viewComponentHelper", "viewComponentHelper");
+            }
+        }
+
+        private void BuildAttributeDeclarations(TagHelperDescriptor descriptor)
+        {
+            Writer.Write("[")
+              .WriteMethodInvocation(typeof(HtmlAttributeNotBoundAttribute).FullName, false, new string[0])
+              .WriteParameterSeparator()
+              .Write(_context.ViewContextType)
+              .Write("]")
+              .WriteLine();
+
+            Writer.WriteAutoPropertyDeclaration(
+                "public",
+                _context.ViewContextType,
+                _context.ViewContextType);
+
+            foreach (var attribute in descriptor.Attributes)
+            {
+                Writer.WriteAutoPropertyDeclaration("public", attribute.TypeName, attribute.PropertyName);
+            }
+        }
+
+        private void BuildProcessMethodString(TagHelperDescriptor descriptor)
+        {
+            var contextVariable = "context";
+            var outputVariable = "output";
+
+            using (Writer.BuildMethodDeclaration(
+                    $"public override async",
+                    nameof(System.Threading.Tasks.Task),
```

Comment:
`typeof(Task).FullName`. Doing a `nameof` like you currently are will just render `Task` and not its fully qualified name.